File: 백준/Silver/1010. 다리 놓기/다리 놓기.py
from sys import stdin

t = int(stdin.readline())
factorial = [1] + [i for i in range(1, 31)]
for i in range(2, 31):
    factorial[i] *= factorial[i-1]
    
for _ in range(t):
    n, m = map(int, stdin.readline().split())
    print(factorial[m]//(factorial[m-n]*factorial[n]))

File: 백준/Silver/1010. 다리 놓기/다리 놓기.java
import java.io.*;
import java.util.*;
import java.math.BigInteger;

public class Main {
    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int t = Integer.parseInt(br.readLine());
        BigInteger[] factorial = new BigInteger[31];
        factorial[0] = BigInteger.ONE;
        for (int i = 1; i < 31; i++) {
            factorial[i] = BigInteger.valueOf((long)i).multiply(factorial[i-1]);
        }

        for (int i = 0; i < t; i++) {
            int[] nm = Arrays.stream(br.readLine().split(" ")).mapToInt(num -> Integer.parseInt(num)).toArray();
            System.out.println(factorial[nm[1]].divide(factorial[nm[1]-nm[0]].multiply(factorial[nm[0]])).toString());
        }
    }
}